core/tests/puppeteer-acceptance-tests/user-utilities/question-admin-utils.js

```diff
@@ -0,0 +1,317 @@
+
+// "Add Contribution Rights" form elements.
+const addContributorUsernameInput =
+  'input#add-contribution-rights-user-input';
+const addContributonRightsCategorySelect =
+  'select#add-contribution-rights-category-select';
+const addContributionRightsSubmitButton =
+   'button#add-contribution-rights-submit-button';
+
+// "Remove Contribution Rights" form elements.
+const removeContributorUsernameInput =
+  'input#remove-contribution-rights-user-input';
+const removeContributonRightsCategorySelect =
+  'select#remove-contribution-rights-category-select';
+const removeContributionRightsSubmitButton =
+  'button#remove-contribution-rights-submit-button';
+
+module.exports = class QuestionAdmin extends baseUser {
+  /**
+   * Function for navigating to the contributor dashboard admin page.
+   */
+  async navigateToContributorDashboardAdminPage() {
+    await this.goto(contributorDashboardAdminUrl);
+  }
+
+  /**
+   * Function for adding a right of reviewing questions to a user.
+   * @param {string} username - the username of the user.
+   */
+  async addReviewQuestionRights(username) {
+    await this.type(addContributorUsernameInput, username);
+    await this.select(
+      addContributonRightsCategorySelect, reviewQuestionRightValue);
+    await this.clickOn(addContributionRightsSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function for adding a right of submitting questions to a user.
+   * @param {string} username - the username of the user.
+   */
+  async addSubmitQuestionRights(username) {
+    await this.type(addContributorUsernameInput, username);
+    await this.select(
+      addContributonRightsCategorySelect, submitQuestionRightValue);
+    await this.clickOn(addContributionRightsSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function for removng a right of reviewing questions to a user.
+   * @param {string} username - the username of the user.
+   */
+  async removeReviewQuestionRights(username) {
+    await this.type(removeContributorUsernameInput, username);
+    await this.select(
+      removeContributonRightsCategorySelect, reviewQuestionRightValue);
+    await this.clickOn(removeContributionRightsSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function for removing a right of reviewing questions to a user.
+   * @param {string} username - the username of the user.
+   */
+  async removeSubmitQuestionRights(username) {
+    await this.type(removeContributorUsernameInput, username);
+    await this.select(
+      removeContributonRightsCategorySelect, submitQuestionRightValue);
+    await this.clickOn(removeContributionRightsSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function to display reviewing questions rights by role
+   * and check if the user is not displayed as a question reviewer
+   * @param {string} usename - the user expected to not be displayed.
```

Comment:
"username" is misspelled, here and elsewhere.

Reply:
Done